IndexingIT.testUpdateSnapshotStatus fails due to a snapshot with the same name being present in master

The failure looks like:

```
org.elasticsearch.backwards.IndexingIT > testUpdateSnapshotStatus FAILED
    org.elasticsearch.client.ResponseException: method [PUT], host [http://127.0.0.1:39507], URI [/_snapshot/repo/bwc-snapshot?wait_for_completion=true], status line [HTTP/1.1 400 Bad Request]
    {"error":{"root_cause":[{"type":"invalid_snapshot_name_exception","reason":"[repo:bwc-snapshot] Invalid snapshot name [bwc-snapshot], snapshot with the same name already exists"}],"type":"invalid_snapshot_name_exception","reason":"[repo:bwc-snapshot] Invalid snapshot name [bwc-snapshot], snapshot with the same name already exists"},"status":400}
        at __randomizedtesting.SeedInfo.seed([18868B88F75EAF0E:7D69A4C50DAE9DB9]:0)
        at org.elasticsearch.client.RestClient.convertResponse(RestClient.java:253)
        at org.elasticsearch.client.RestClient.performRequest(RestClient.java:231)
        at org.elasticsearch.client.RestClient.performRequest(RestClient.java:237)
        at org.elasticsearch.client.RestClient.performRequest(RestClient.java:205)
        at org.elasticsearch.backwards.IndexingIT.testUpdateSnapshotStatus(IndexingIT.java:267)
```

The reproduction line is:

```
./gradlew ':qa:mixed-cluster:v7.5.0#mixedClusterTest' --tests "org.elasticsearch.backwards.IndexingIT.testUpdateSnapshotStatus" -Dtests.seed=18868B88F75EAF0E -Dtests.security.manager=true -Dtests.locale=gsw -Dtests.timezone=Etc/GMT+12 -Dcompiler.java=12 -Druntime.java=11
```

But it did not reproduce for me.

https://elasticsearch-ci.elastic.co/job/elastic+elasticsearch+master+multijob+fast+bwc/1496/consoleFull
https://gradle-enterprise.elastic.co/s/34ngdnkksq3ce

This is an actual bug, tripping an assertion in a BwC test:

```
» ERROR][o.e.b.ElasticsearchUncaughtExceptionHandler] [v7.5.0-0] fatal error in thread [elasticsearch[v7.5.0-0][generic][T#3]], exiting
»  java.lang.AssertionError: Received null generation for shard state [SUCCESS]
```

on it

--- a/server/src/main/java/org/elasticsearch/cluster/SnapshotsInProgress.java
+++ b/server/src/main/java/org/elasticsearch/cluster/SnapshotsInProgress.java
@@ -279,7 +279,6 @@ public class SnapshotsInProgress extends AbstractNamedDiffable<Custom> implement
             state = ShardState.fromValue(in.readByte());
             if (in.getVersion().onOrAfter(SnapshotsService.SHARD_GEN_IN_REPO_DATA_VERSION)) {
                 generation = in.readOptionalString();
-                assert generation != null || state != ShardState.SUCCESS : "Received null generation for shard state [" + state + "]";
             } else {
                 generation = null;
             }

